backend/src/main/java/de/neuefische/backend/controller/RecipeController.java

```diff
@@ -18,4 +18,16 @@ public class RecipeController {
     public List<RecipeNormalized> getAllRecipes(){
         return service.getAllRecipes();
     }
+
+    @GetMapping("/{id}")
+    public RecipeNormalized getRecipeById(@PathVariable String id){
+        return service.getRecipeById(id);
+    }
+
+    @ExceptionHandler(RecipeNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public String handleRecipeNotFoundException(RecipeNotFoundException exception){
+        return exception.getMessage();
+    }
```

Comment:
It will get a bit more readable, if you use a GlobalExceptionHandler for handling those, instead of having your RestController run double duty.

Reply:
Great suggestion! done!